### PR TITLE
:as-is-paths to skip value conversion for specific properties

### PR DESCRIPTION
**Pull request by codeperfector**

### CODE REVIEW

File: src/cprop/source.cljc [hunk 1]
```diff
@@ -69,8 +69,11 @@
    (read-system-env {}))
   ([opts]
    (->> (System/getenv)
-        (map (fn [[k v]] [(env->path k opts)
-                          (str->value v opts)]))
+        (map (fn [[k v]]
+               (let [path        (env->path k opts)
+                     as-is-paths (:as-is-paths opts)]
+                   [path (->> (assoc opts :as-is? (or (:as-is? opts) (and as-is-paths (as-is-paths path))))
```
Comment (tolitius) [inline src/cprop/source.cljc:75]: this is repeated 3 times in 3 functions
let's refactor it out into it's own function that takes opts, and path and returns opts

```clojure
(defn with-as-is [{:keys [as-is?
                          as-is-paths] :as opts} path]
  (assoc opts :as-is? (or as-is?
                          (and as-is-paths
                               (as-is-paths path))))
```

then the let block for all the places this is used would look somm like:

```clojure
(let [path (sysprop->path k opts)]
  [path (->> (with-as-is opts path)
             (str->value v))])
```

**Review by tolitius — CHANGES_REQUESTED** [review]
thanks for the contribution!
I left a comment to refactor logic into a function

**Comment by codeperfector** [comment]
I performed the changes requested. I also added a README section for this feature.

File: README.md [hunk 2]
```diff
@@ -653,6 +653,12 @@ If you need _ALL_ the properties and configs to come in "as is" (not as EDN) `:a
 (load-config :as-is? true)
 ```
 
+You can also opt to keep value conversion for all your properties with the exception of a set of specified paths using `:as-is-paths`:
+
+```clojure
+(load-config :as-is-paths #{'(:io :http :pool :socket-timeout)})
```
Comment (tolitius) [inline README.md:659]: for sequences, in Clojure, lists "`'()`" are rarely used in APIs, and vectors are preferred
lists are good to you when you actually need a linked list semantics: i.e. parsing code, append, etc.

so the above example would be:

```clojure
(load-config :as-is-paths #{[:io :http :pool :socket-timeout]})
```

also, in order to convey the message of these paths being a set it is good to give a few paths as an example:

```clojure
(load-config :as-is-paths #{[:io :http :pool :socket-timeout]
                            [:datomic :max-conn]
                            [:some :other :path]})
```

File: src/cprop/source.cljc [hunk 3]
```diff
@@ -56,6 +56,12 @@
          (catch Throwable _
            v))))
 
+(defn ^:private with-as-is
+    [{:keys [as-is? as-is-paths] :as opts} path]
+    (assoc opts :as-is? (or as-is?
+                            (and as-is-paths
+                                 (as-is-paths path)))))
```
Comment (tolitius) [inline src/cprop/source.cljc:63]: format as (two spaces below the form):
```clojure
(defn ^:private with-as-is
  [{:keys [as-is? as-is-paths] :as opts} path]
  (assoc opts :as-is? (or as-is?
                          (and as-is-paths
                               (as-is-paths path)))))
```
so it is consistent with the codebase

File: src/cprop/source.cljc [hunk 4]
```diff
@@ -69,8 +75,10 @@
    (read-system-env {}))
   ([opts]
    (->> (System/getenv)
-        (map (fn [[k v]] [(env->path k opts)
-                          (str->value v opts)]))
+        (map (fn [[k v]]
+               (let [path        (env->path k opts)]
+                   [path (->> (with-as-is opts path)
+                              (str->value v))])))
```
Comment (tolitius) [inline src/cprop/source.cljc:81]: format:
```clojure
(map (fn [[k v]]
       (let [path (env->path k opts)]
         [path (->> (with-as-is opts path)
                    (str->value v))])))
```

File: src/cprop/source.cljc [hunk 5]
```diff
@@ -88,8 +96,10 @@
    (read-system-props {}))
   ([opts]
    (->> (System/getProperties)
-        (map (fn [[k v]] [(sysprop->path k opts)
-                          (str->value v opts)]))
+        (map (fn [[k v]]
+               (let [path        (sysprop->path k opts)]
+                   [path (->> (with-as-is opts path)
+                              (str->value v))])))
```
Comment (tolitius) [inline src/cprop/source.cljc:102]: format:
```clojure
(map (fn [[k v]]
       (let [path (sysprop->path k opts)]
         [path (->> (with-as-is opts path)
                    (str->value v))])))
```

File: src/cprop/source.cljc [hunk 6]
```diff
@@ -119,11 +129,12 @@
...
-                         (str->value (if-not (false? parse-seqs?) ;; could be nil, which is true in this case
-                                       (prop-seq v)
-                                       v)
-                                     opts)]))
+       (map (fn [[k v]]
+              (let [prop-path   (prop-key->path k opts)]
+                  [prop-path (->> (with-as-is opts prop-path)
+                                  (str->value (if-not (false? parse-seqs?) ;; could be nil, which is true in this case
+                                                  (prop-seq v)
+                                                  v)))])))
```
Comment (tolitius) [inline src/cprop/source.cljc:137]: format:
```clojure
(map (fn [[k v]]
       (let [prop-path (prop-key->path k opts)]
         [prop-path (->> (with-as-is opts prop-path)
                         (str->value (if-not (false? parse-seqs?) ;; could be nil, which is true in this case
                                       (prop-seq v)
                                       v)))])))
```

File: test/cprop/test/core.cljc [hunk 7]
```diff
@@ -249,6 +249,7 @@
                        (System/setProperty k v))
         props        (from-system-props)
         props-as-is  (from-system-props {:as-is? true})
+        props-as-is-path  (from-system-props {:as-is-paths #{'(:io :http :pool :socket-timeout)}})
```
Comment (tolitius) [inline test/cprop/test/core.cljc:252]: do add cases with a few paths to make sure it works with sets of paths
also `'()` to `[]` for paths

**Review by tolitius — CHANGES_REQUESTED** [review]
thanks for the changes
I added a few cleanup comments

**Comment by codeperfector** [comment]
@tolitius Thank you. I made the changes you requested.

**Comment by tolitius** [comment]
@codeperfector 
thanks, looks great!

**Pull request merged**